Sources/swift-format/main.swift

```diff
@@ -69,18 +69,36 @@ fileprivate func main(_ arguments: [String]) -> Int32 {
 ///   - transform: A closure that performs a transformation on a specific source file.
 private func processSources(
   from paths: [String], configurationPath: String?,
-  transform: (FileHandle, String, Configuration) -> Int
+  transform: @escaping (FileHandle, String, Configuration) -> Int
 ) -> Int32 {
   var result = 0
+  let concurrentQueue = DispatchQueue(label: "com.queue.Concurrent", qos: .userInitiated, attributes: .concurrent)
+  let group = DispatchGroup()
+  let lock = NSLock()
   for path in FileIterator(paths: paths) {
-    guard let sourceFile = FileHandle(forReadingAtPath: path) else {
-      stderrStream.write("Unable to create a file handle for source from \(path).\n")
-      stderrStream.flush()
-      return 1
+    group.enter()
+    concurrentQueue.async {
+      defer { group.leave() }
+      guard let sourceFile = FileHandle(forReadingAtPath: path) else {
+        stderrStream.write("Unable to create a file handle for source from \(path).\n")
+        stderrStream.flush()
+        lock.lock()
+        result |= 1
+        lock.unlock()
+        return
+      }
+
+      let configuration = loadConfiguration(forSwiftFile: path, configFilePath: configurationPath)
+      let transformResult = transform(sourceFile, path, configuration)
```

Comment:
It looks like there is a way to make it write atomically https://github.com/apple/swift-tools-support-core/blob/e11ef6f91b3bfcca36bda9b0f0cc38876328db78/Sources/TSCBasic/FileSystem.swift#L378-L395 although right now I believe this is broken since it appears `moveItem` doesn't actually allow overwriting:

```
Error Domain=NSCocoaErrorDomain Code=516 "“foo.txt” couldn’t be moved to “swift-format” because an item with the same name already exists." UserInfo={NSSourceFilePathErrorKey=/Users/ksmiley/dev/oss-swift/swift-format/foo.txt, NSUserStringVariant=(
    Move
), NSDestinationFilePath=/Users/ksmiley/dev/oss-swift/swift-format/bar.txt, NSFilePath=/Users/ksmiley/dev/oss-swift/swift-format/foo.txt, NSUnderlyingError=0x7fd475d19f20 {Error Domain=NSPOSIXErrorDomain Code=17 "File exists"}}
```

Reply:
I submitted https://github.com/apple/swift-tools-support-core/pull/56 to fix this